### PR TITLE
770 add footer contributions catalog

### LINKED ISSUE #770: [FEATURE] Add footer to Contributions Catalog
**Is your feature request related to a problem? Please describe.**
Currently, we don't have a footer in the Contributions Catalog.

**Describe the solution you'd like**
Please add a footer to the Contributions Catalog. This could be implemented using a base layout. The following details that are separated by a vertical bar (`|`) need to be added to the footer :

1. Catalog version number (e.g., v1.10.11). This can be obtained by parsing the latest git release tag. 
2. © <current year> University of Illinois Board of Trustees
2. Privacy Policy that links to https://www.vpaa.uillinois.edu/resources/web_privacy

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by minump**
## Description
Used GitPython Repo to get latest tag(with respect to commit date). This tag is populated when used outside of flask app. However, this is not being populated when used inside of flask method index().
Updated requirements to add GitPython
Updated catalog_rest_service.py to fetch tag using GitPython Repo.
Updated home.html to render the git tag.

**Fixes #770**

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Immediately
- [x] Within a week
- [ ] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [x] I have updated the [CHANGELOG](../CHANGELOG.md).
- [x] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

### CODE REVIEW

**Comment by minump** [comment]
@sandeep-ps putting as a draft PR as I am facing issues in rendering git tag from catalog_rest_service to home.html.

**Comment by sandeep-ps** [comment]
> @sandeep-ps putting as a draft PR as I am facing issues in rendering git tag from catalog_rest_service to home.html.

Okay. Are you able to print the tag in the Python endpoint? Could you please also include the footer in other pages like contribute, edit, etc? I will definitely take a look at this today afternoon.

**Comment by minump** [comment]
> > @sandeep-ps putting as a draft PR as I am facing issues in rendering git tag from catalog_rest_service to home.html.
> 
> Okay. Are you able to print the tag in the Python endpoint? Could you please also include the footer in other pages like contribute, edit, etc? I will definitely take a look at this today afternoon.

I am able to print the gittag outside of the index() function.
Will make changes to include footer in other webpages too.

**Comment by minump** [comment]
Able to display gittag in html page for flask tutorial. Unsure why the same code is not rendering gittag in contributions catal
Screen
Screen Shot 2021-09-20 at 11 32 32 AM
 Shot 2021-09-20 at 11 31 32 AM
og

**Comment by minump** [comment]
Screen Shot 2021-09-20 at 11 32 32 AM

**Comment by sandeep-ps** [comment]
I think I see what the problem is. home.html is being rendered using two different endpoints in two files. You can see the other one in contribute.py. If you go to http://localhost:5050/, it's being displayed. It's not being displayed for http://localhost:5050/catalog/. I think it should be a matter of adding the version to the other endpoint too. You may want to consider, storing the value in a common variable and then reusing it for multiple endpoints. I hope it helps. Please let me know if you have any other questions. Thanks.

**Comment by minump** [comment]
Added gittag to contribute.py. Footer rendered in home.html.

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,11 +23,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Contribution POST gives proper error message. [#714](https://github.com/rokwire/rokwire-building-blocks-api/issues/714)
 - Hard coded base url. [#780](https://github.com/rokwire/rokwire-building-blocks-api/issues/780)
 - Multiple bugs/improvements around login. [#782](https://github.com/rokwire/rokwire-building-blocks-api/issues/782)
+<<<<<<< HEAD
```
Comment (sandeep-ps) [inline CHANGELOG.md:26]: Please remove the merge conflict tags in CHANGELOG.

File: contributions/catalog/webapps/templates/contribute/home.html [hunk 2]
```diff
@@ -215,6 +226,9 @@ <h2 id="title"> Contributions Catalog/Packager </h2>
             <button class="btn" id="talent_btn" onclick="get_url('talent')"> Talents</button>
         {% endif %}
 
+        <div class="footer">
+            <p>Catalog version {{ gittag }} &#124; &copy; University of Illinois Board of Trustees &#124; <a href="https://www.vpaa.uillinois.edu/resources/web_privacy">Privacy Policy</a> </p>
```
Comment (sandeep-ps) [inline contributions/catalog/webapps/templates/contribute/home.html:230]:
```suggestion
            <p>Version {{ gittag }} &emsp; &#124; &emsp; &copy; University of Illinois Board of Trustees &emsp; &#124; &emsp; <a href="https://www.vpaa.uillinois.edu/resources/web_privacy">Privacy Policy</a> </p>
```
Suggesting some UI changes.

**Comment by minump** [comment]
resolved conflicts in changelog. updated some UI changes to the footer in home.html

**Comment by sandeep-ps** [comment]
Please fix the merge conflict in the CHANGELOG. Thanks.

**Comment by minump** [comment]
> Please fix the merge conflict in the CHANGELOG. Thanks.

Apologize for not checking this earlier. Conflicts resolved.

**Comment by sandeep-ps** [comment]
> > Please fix the merge conflict in the CHANGELOG. Thanks.
> 
> Apologize for not checking this earlier. Conflicts resolved.

No need to apologize. This conflict came up only after merging the previous PR. :-)

**Comment by sandeep-ps** [comment]
If it's not too late, I think the release script in the root folder might be used to calculate git tag and inject using build arg as Rob suggested. New Docker instructions will need to added to the README in that case. Thanks.

**Comment by minump** [comment]
> If it's not too late, I think the [release script](https://github.com/rokwire/rokwire-building-blocks-api/blob/develop/release_contributions_catalog.sh) in the root folder might be used to calculate git tag and inject using build arg as Rob suggested. New Docker instructions will need to added to the README in that case. Thanks.

Will update release script to get git tag and git sha and pass as docker build args. In python code, check if the arg exists, and if not, display git tag and git sha as empty string.

**Comment by minump** [comment]
Changed PR to draft as release_contributions_catalog.sh run contributions/Dockerfile instead of catalog/Dockerfile. Unsure if I'm missing anything.
Made neccesary changes to release scripts and Dockerfile.

**Comment by sandeep-ps** [comment]
I have merged #801 to fix the release script. You continue with reopening this PR. Thanks.

File: contributions/catalog/catalog_rest_service.py [hunk 3]
```diff
@@ -113,7 +125,8 @@ def index():
             cap_json = jsonutil.create_capability_json_from_contribution_json(result.json())
             tal_json = jsonutil.create_talent_json_from_contribution_json(result.json())
 
-    return render_template('contribute/home.html', cap_json=cap_json, tal_json=tal_json, show_sel=show_sel, user=user)
+    return render_template('contribute/home.html', gittag=gittag, gitsha=gitsha,
```
Comment (minump) [inline contributions/catalog/catalog_rest_service.py:128]: the render_template('contribute/home.html') returns template not found error if running 
```
python contributions/catalog/catalog_rest_service.py
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -432,4 +432,4 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 [1.0.3]: https://github.com/rokwire/rokwire-building-blocks-api/compare/1.0.2...1.0.3
 [1.0.2]: https://github.com/rokwire/rokwire-building-blocks-api/compare/1.0.1...1.0.2
 [1.0.1]: https://github.com/rokwire/rokwire-building-blocks-api/compare/1.0.0...1.0.1
-[1.0.0]: https://github.com/rokwire/rokwire-building-blocks-api/releases/tag/1.0.0
+[1.0.0]: https://github.com/rokwire/rokwire-building-blocks-api/releases/tag/1.0.0
```
Comment (sandeep-ps) [inline CHANGELOG.md:435]: I think this particular change is not needed. Please update the CHANGELOG with an entry for this task. Thanks.

**Review by sandeep-ps — APPROVED** [review]

**Comment by sandeep-ps** [comment]
Thanks for your feedback, @robkooper. 

@ywkim312 or @robkooper,  I was planning to merge this PR today afternoon. Any final comments?

**Comment by ywkim312** [comment]
> Thanks for your feedback, @robkooper.
> 
> @ywkim312 or @robkooper, I was planning to merge this PR today afternoon. Any final comments?

no, let's merge

**Review by robkooper — APPROVED** [review]

**Comment by sandeep-ps** [comment]
Thank you, All.

**Pull request merged**